### ISSUE TITLE
constexpr macro parsing issue

### ISSUE THREAD

**Issue by hashinclude** [tree-sitter/tree-sitter-cpp#87]
Declaring a constexpr macro works well if the total length of the line is less than 38 chars -  
<img width="1680" alt="Screen Shot 2020-08-13 at 5 04 42 AM" src="https://user-images.githubusercontent.com/1989695/90132354-80567e00-dd22-11ea-911d-c2d5ba1bf7a9.png">

However, adding one more char to the line containing the macro causes the following class definition to be incorrectly handled. 
<img width="1680" alt="Screen Shot 2020-08-13 at 5 06 46 AM" src="https://user-images.githubusercontent.com/1989695/90132480-cb709100-dd22-11ea-9b7a-8fa9d45b179f.png">

**Comment by maxbrunsfeld** [comment]
Yeah, the error recovery heuristics take into account the *size* of text that must be wrapped in an error (in addition to the number of syntax nodes), so it is going to depend on the text of the code that's causing the parse error.
